src/module.ts

```diff
@@ -0,0 +1,193 @@
+    nuxt.options.build.transpile.push(runtimeDir)
+
+    // Add plugins
+    for (const template of ['plugin.client', 'plugin.server']) {
+      addPlugin(resolve(runtimeDir, template))
+    }
+
+    addComponent({ name: options.componentName, filePath: resolve(runtimeDir, 'component.vue') })
+
+    // Nuxt 3 - SSR false
+    // TODO: use nitro hooks
+    if (!nuxt.options.ssr) {
+      nuxt.hook('nitro:document', (template) => {
+        template.contents = template.contents.replace('</body>', `</body><script>${options.script}</script>`)
+      })
+    }
+
+    // Nuxt 2 - SSR false
+    nuxt.hook('vue-renderer:spa:prepareContext', ({ head }) => {
+      const script = {
+        hid: options.hid,
+        innerHTML: options.script,
+        pbody: true
+      }
+
+      head.script.push(script)
+
+      const serializeProp = '__dangerouslyDisableSanitizersByTagID'
+      head[serializeProp] = head[serializeProp] || {}
+      head[serializeProp][options.hid] = ['innerHTML']
+    })
+
+    // Nuxt 2 - SSR true
+    nuxt.hook('vue-renderer:ssr:csp', (cspScriptSrcHashes) => {
+      const { csp } = nuxt.options.render
+      const hash = crypto.createHash((csp as any).hashAlgorithm)
+      hash.update(options.script!)
+      cspScriptSrcHashes.push(
+        `'${(csp as any).hashAlgorithm}-${hash.digest('base64')}'`
+      )
+    })
+
+    // In Nuxt 2 dev mode we also inject full script via webpack entrypoint for storybook compatibility
+    if (nuxt.options.dev && isNuxt2()) {
+      const { dst } = addTemplate({
+        src: resolve(__dirname, 'script.min.js'),
+        fileName: join('color-mode', 'script.min.js'),
+        options
+      })
+      nuxt.hook('webpack:config', (configs) => {
+        for (const config of configs) {
+          if (config.name !== 'server') {
+            (config.entry as any).app.unshift(resolve(nuxt.options.buildDir, dst!))
+          }
+        }
+      })
+    }
+  }
+})
+
+export interface ColorModeOptions {
+  /**
+   * Default: `system`
+   */
+  preference: string, // default value of $colorMode.preference
+  /**
+   * Default: `light`
+   */
+  fallback: string, // fallback value if not system preference found
+  /**
+   * Default: `nuxt-color-mode-script`
+   */
+  hid: string,
+  /**
+   * Default: `__NUXT_COLOR_MODE__`
+   */
+  globalName: string,
+  /**
+   * Default: `ColorScheme`
+   */
+  componentName: string,
+  /**
+   * Default: ''
+   */
+  classPrefix: string,
+  /**
+   * Default: '-mode'
+   */
+  classSuffix: string,
+  /**
+   * Default: 'nuxt-color-mode'
+   */
+  storageKey: string
+  /**
+   * The script that will be injected into the head of the page
+   */
+  script?: string
+}
+
+export type ColorModeConfig = Partial<ColorModeOptions>
+export type ModuleOptions = ColorModeConfig
+
+// @ts-ignore
+declare module '@nuxt/types/config/index' {
+  interface NuxtOptions {
+    colorMode: ColorModeConfig
+  }
+}
+
+// @ts-ignore
+declare module '@nuxt/vue-app' {
```

Comment:
(same refactor suggestion for `@nuxt/types`, `@nuxt/vue-app`. Nuxt 2 without bridge should be unsupported IMO in next rewrite of modules.